semantic-analysis/src/Analysis/Effect/Domain.hs

```diff
@@ -79,6 +81,15 @@ dstring :: Has (Dom val) sig m => Text -> m val
 dstring = send . DString
 
 
+-- Statements
+
+-- | Combine the results of adjacent statements.
+--
+-- This exists to allow e.g. collecting analyses to see the results of earlier statements when processing later ones, without requiring @val@ to be a 'Semigroup'. For example, concrete and typechecking domains would likely ignore the first parameter and return the second, while a domain counting the number of visited instructions would return the sum of both.
+(>>>) :: Has (Dom val) sig m => val -> val -> m val
```

Comment:
Collision with the one from `Control.Category` here, but I’m not stressed about it.

Reply:
I'd have preferred `>>` but that's even more intrusive a collision.